internal/lang/evalchecks/eval_lifecycle_enabled_test.go

```diff
@@ -0,0 +1,190 @@
+			[]WantedError{
+				{
+					"Invalid enabled argument",
+					`The given "enabled" argument value is unsuitable: bool required, but have number.`,
+				},
+				{
+					"Invalid enabled argument",
+					`The given "enabled" argument value is unsuitable: the given value is null.`,
+				},
+			},
+		},
+		"1": {
+			cty.NumberIntVal(1),
+			[]WantedError{
+				{
+					"Invalid enabled argument",
+					`The given "enabled" argument value is unsuitable: bool required, but have number.`,
+				},
+			},
+		},
+		"negative": {
+			cty.NumberIntVal(-1),
+			[]WantedError{
+				{
+					"Invalid enabled argument",
+					`The given "enabled" argument value is unsuitable: bool required, but have number.`,
+				},
+			},
+		},
+		"list": {
+			cty.ListVal([]cty.Value{cty.StringVal("a"), cty.StringVal("a")}),
+			[]WantedError{
+				{
+					"Invalid enabled argument",
+					"The given \"enabled\" argument value is unsuitable: bool required, but have list of string.",
+				},
+			},
+		},
+		"tuple": {
+			cty.TupleVal([]cty.Value{cty.StringVal("a"), cty.StringVal("b")}),
+			[]WantedError{
+				{
+					"Invalid enabled argument",
+					"The given \"enabled\" argument value is unsuitable: bool required, but have tuple.",
+				},
+			},
+		},
+		"unknown": {
+			cty.UnknownVal(cty.Number),
+			[]WantedError{
+				{
+					"Invalid enabled argument",
+					"The given \"enabled\" argument value is unsuitable: bool required, but have number.",
+				},
+				{
+					"Invalid enabled argument",
+					`The given "enabled" argument value is derived from a value that won't be known until the apply phase, so OpenTofu cannot determine whether an instance of this object is declared or not.`,
+				},
+			},
+		},
+		"sensitive": {
+			cty.StringVal("1").Mark(marks.Sensitive),
+			[]WantedError{
+				{
+					"Invalid enabled argument",
+					`Sensitive values, or values derived from sensitive values, cannot be used in "enabled" arguments. If used, the sensitive value would be exposed by whether an instance is present.`,
+				},
+			},
+		},
+		"ephemeral": {
+			cty.StringVal("1").Mark(marks.Ephemeral),
+			[]WantedError{
+				{
+					"Invalid enabled argument",
+					`The given "enabled" argument value is unsuitable: the given value is ephemeral.`,
+				},
+			},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			_, diags := EvaluateEnabledExpression(hcltest.MockExprLiteral(test.val), mockRefsFunc())
+
+			if len(diags) != len(test.Wanted) {
+				t.Fatalf("wrong diagnostics size: (want %d, got %d):\n", len(test.Wanted), len(diags))
+			}
+
+			for i, wanted := range test.Wanted {
+				diag := diags[i]
+				if diff := cmp.Diff(diag.Description().Summary, wanted.Summary); diff != "" {
+					t.Errorf("%d: wrong summary (-want +got):\n%s", i, diff)
+				}
+
+				if diff := cmp.Diff(diag.Description().Detail, wanted.DetailSubstring); diff != "" {
+					t.Errorf("%d: wrong description (-want +got):\n%s", i, diff)
+				}
+			}
```

Comment:
Wouldn't be easier to convert the diagnostics received in a slice named `gotDiagsInfo` and then run `cmp.Diff(test.Wanted, gotDiagsInfo)`